### PR TITLE
Include unittest to create chain with txns

### PR DESCRIPTION
**Pull request by viquezclaudio**
Created a new unittest that can be used to exercise a chain
with transactions indefinitely

### CODE REVIEW

File: block-production/Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ maintenance = { status = "experimental" }
 log = "0.4"
 hex = "0.4"
 parking_lot = { git = "https://github.com/styppo/parking_lot.git" }
+rand = "0.8"
```
Comment (jsdanielh) [inline block-production/Cargo.toml]: Maybe a `dev` dependency instad?

File: block-production/tests/mod.rs [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
 use nimiq_test_utils::blockchain::{
-    fill_micro_blocks, sign_macro_block, sign_view_change, signing_key, voting_key,
+    fill_micro_blocks, fill_micro_blocks_with_txns, sign_macro_block, sign_view_change,
+    signing_key, voting_key,
 };
 use nimiq_transaction_builder::TransactionBuilder;
 use nimiq_utils::time::OffsetTime;
 use nimiq_vrf::VrfSeed;
+use parking_lot::RwLock;
+use tempdir::TempDir;
```
Comment (jsdanielh) [inline block-production/tests/mod.rs]: We normally separate internal vs external dependencies... I would put them above as they were

File: test-utils/src/blockchain.rs [hunk 3]
```diff
@@ -1,5 +1,8 @@
+use std::str::FromStr;
 use std::sync::Arc;
+use std::time::Instant;
 
+use nimiq_primitives::coin::Coin;
```
Comment (jsdanielh) [inline test-utils/src/blockchain.rs]: I would place them below with nimiq dependencies

File: block-production/tests/mod.rs [hunk 4]
```diff
@@ -27,6 +31,8 @@ pub const ACCOUNT_SECRET_KEY: &str =
 
 const STAKER_ADDRESS: &str = "NQ20TSB0DFSMUH9C15GQGAGJTTE4D3MA859E";
 
+const VOLATILE_ENV: bool = false;
```
Comment (jsdanielh) [inline block-production/tests/mod.rs]: I would consider leaving the default in `true` since most of our tests run in a volatile environment

**Review by jsdanielh — CHANGES_REQUESTED** [review]

**Comment by viquezclaudio** [comment]
changes applied @jsdanielh

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**